src/org/daisy/dotify/api/formatter/Context.java

```diff
@@ -58,6 +58,20 @@ public default Integer getMetaVolume() {
 	public default Integer getMetaPage() {
 		return null;
 	}
+    
+	/**
+	 * Gets the page number (one based) of the first page of the content of the
+     * context described in the current context, or null if not available.
+	 *  
+	 * This method can be used in a toc-entry-on-resumed element to retrieve the
+     * first page number of the volume's content in the original context.
```

Comment:
I know you didn't originally write this, but I found this a bit hard to understand. Don't you?

While trying to comprehend it, I realized that the intention of "metaPage" is to be a generic field that can be used for whatever makes sense in a certain context. Currently it is used for the `$started-page-number` variable in the context of a `on-page-start` or `on-page-end` element (in `dynamic-sequence`). The documentation of `getMetaPage` also suggests that it could be used to contain the page number of what is being referenced in the context of a `toc-entry` or `item` element, however this is currently not implemented. (To insert page numbers in a table of contents we use the `page-number` element, and for end notes we don't put page numbers in the note itself, only before/after a group of notes, using `on-page-start`/`on-page-end`.)

In the context of a `toc-entry-on-resumed`, the thing that makes most sense to put in this generic meta field is exactly ... the `$started-volume-first-content-page` variable. In other words, we don't need a new field, because the existing one already covers it. If we want to add a specialized field for each use case, we don't need the generic one anymore, and we'd have to add one for the `$started-page-number` case, and for every new variable that we come up with. For example I think it would be nice to have a variable to point to the referenced page in a `toc-entry`. If that variable would exist we would be able to deprecate the `page-number` element. Simply using the single generic field for all these variables, like Joel intended, seems the easier solution.

Reply:
The Javadoc could be improved like this:

```java
/**
 * Gets the page number (one based) of the context described in the current context, or null if
 * not available.
 *
 * <ul>
 *   <li>For example, in the context of an entry in a table of contents (<code>toc-entry</code> in
 *   OBFL), this method can be used to retrieve the page number of the block the entry is connected
 *   with.</li>
 *   <li>In case of a <code>toc-entry-on-resumed</code> the method would be used to retrieve the
 *   number of the first content page after the volume break that corresponds with this entry.</li>
 *   <li>In the context of an end note (<code>item</code> in OBFL) this method can be used to
 *   retrieve the page number of the anchor that the item is connected with.</li>
 *   <li>In the context of a group of end notes that are referenced from the same page
 *   (<code>on-page-start</code> and <code>on-page-end</code> in OBFL), this method can be used to
 *   retrieve the number of that page.</li>
 *   <li>Etc.</li>
 * </ul>
 */
public Integer getMetaPage();
```

Reply:
>In the context of a toc-entry-on-resumed, the thing that makes most sense to put in this generic meta field is exactly ... the $started-volume-first-content-page variable. In other words, we don't need a new field, because the existing one already covers it.

I didn't realize that in the context of a `toc-entry-on-resumed` the `meta-page` did not already have a meaningful value. It makes sense to use the existing `meta-page` for the `$started-volume-first-content-page` variable. I will make these changes.

Reply:
OK cool. Just to be absolutely clear, `getMetaPage` does not correspond with a single variable. It is just container for _some_ variable value. In the implementation the variable name will have to be set depending on the context (see the `OBFLExpressionBase.setMetaPageNumberVariable()` method).

Reply:
Done.

@bertfrees @kalaspuffar  Please review this pull request again.